cmd/apiserver-watcher/run.go

```diff
@@ -67,10 +62,28 @@ func runRunCmd(cmd *cobra.Command, args []string) error {
 		return fmt.Errorf("invalid URI %q (no scheme)", uri)
 	}
 
-	handler, err := newHandler(uri)
-	if err != nil {
-		return err
+	// Get the VIP
+	var vip string
+	if runOpts.vip != "" {
+		vip = runOpts.vip
+		if net.ParseIP(vip) == nil {
+			return fmt.Errorf("vip %s is an invalid IP address", vip)
+		}
+		if utilnet.IsIPv6String(vip) != runOpts.ipv6 {
+			return fmt.Errorf("vip %s IP family mismatch, ipv6 option %v", vip, runOpts.ipv6)
+		}
+	} else {
+		addrsList, err := net.LookupHost(uri.Hostname())
+		if err != nil {
+			return fmt.Errorf("failed to lookup host %s: %v", uri.Hostname(), err)
+		}
+		addrs := filterAddrList(addrsList, runOpts.ipv6)
```

Comment:
This solves the problem with dualstack urls, before it checked it only returned 1 address for all families, breaking dualstack.
This keeps the same restriction of 1 address, but for only one IP family

Reply:
could we add this as a comment here bc i think that would be helpful